### PR TITLE
feat: convert two callbacks into one(PRLC100-343)

### PR DESCRIPTION
**Pull request by rahulsanjava19**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -538,8 +538,12 @@ def sonarExclusions = [
   '**uk/gov/hmcts/reform/prl/config/**/*',
   '**uk/gov/hmcts/reform/prl/utils/*',
   '**uk/gov/hmcts/reform/prl/Application.*',
-  '**uk/gov/hmcts/reform/prl/controllers/citizen/CaseController.java',
-  '**uk/gov/hmcts/reform/prl/controllers/citizen/CitizenCallbackController.java'
+  '**uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMapper.java',
```
Comment (shashisk97) [inline build.gradle:541]: Can we please remove these exclusions?

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMapper.java [hunk 2]
```diff
@@ -47,81 +48,83 @@ public CaseData buildUpdatedCaseData(CaseData caseData) throws JsonProcessingExc
         ObjectMapper mapper = new ObjectMapper();
         CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder();
 
-        if (isNotEmpty(caseData.getC100RebuildInternationalElements())) {
+        C100RebuildData c100RebuildData = caseData.getC100RebuildData();
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMapper.java:51]: Why CaseDataMapper is under controllers package?

**Review by shashisk97 — CHANGES_REQUESTED** [review]
There are 20 code smells as well, which needs to be fixed

File: config/owasp/suppressions.xml [hunk 3]
```diff
@@ -27,6 +27,7 @@
         <cve>CVE-2022-38749</cve>
         <cve>CVE-2022-38752</cve>
         <cve>CVE-2022-42252</cve>
+        <cve>CVE-2021-37533</cve>
```
Comment (AlokDatta) [inline config/owasp/suppressions.xml:30]: Dont think we need this

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java [hunk 4]
```diff
@@ -237,6 +238,13 @@ public AboutToStartOrSubmitCallbackResponse generateDocumentSubmitApplication(
         caseDataUpdated.putAll(caseSummaryTab.updateTab(caseData));
         caseDataUpdated.putAll(map);
 
+        if (null != caseData.getCaseCreatedBy() && caseData.getCaseCreatedBy().equals(CaseCreatedBy.CITIZEN)) {
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java:241]: Revert the equation. you wont need null check

Comment (rahulsanjava19) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java]: Removed

File: config/owasp/suppressions.xml (hunk 3 above)
Comment (rahulsanjava19) [reply config/owasp/suppressions.xml]: Removed

File: build.gradle (hunk 1 above)
Comment (rahulsanjava19) [reply build.gradle:541]: will be covered https://tools.hmcts.net/jira/browse/PRLC100-429

**Review by shashisk97 — APPROVED** [review]
LGTM

**Pull request merged**